posts/views.py

```diff
@@ -34,6 +36,17 @@ def permission_denied(self, request, message=None, code=None):
     """
     raise exceptions.PermissionDenied(detail=message, code=code)
   
+  def list(self, request, *args, **kwargs):
+    # It does not include private posts from other users.
+    queryset = self.filter_queryset(self.get_queryset()).exclude(~Q(author=request.user)&Q(is_public=False))
+    page = self.paginate_queryset(queryset)
+    if page is not None:
+        serializer = self.get_serializer(page, many=True)
+        return self.get_paginated_response(serializer.data)
+
+    serializer = self.get_serializer(queryset, many=True)
+    return Response(serializer.data)
+  
```

Comment:
list메서드 전체를 오버라이딩 할 필요없이 list메서드에서 사용하는 쿼리셋만 수정하면 됨.
```suggestion
  def list(self, request, *args, **kwargs):
    # It does not include private posts from other users.
    self.queryset = self.queryset.exclude(~Q(author=request.user)&Q(is_public=False))
    return super().list(request, args, kwargs)
```